python/hsfs/engine/python.py

```diff
@@ -240,7 +250,7 @@ def _read_s3(self, storage_connector, location, data_format):
         return df_list
 
     def read_options(self, data_format, provided_options):
-        return {}
+        return provided_options or {}
```

Comment:
Isn't this going to return `True` or `False` since it's a logical expression?
Shouldn't it be something like this
```suggestion
        return provided_options if provided_options else {}
```

Reply:
It will return provided_options if provided_options is True (not empty)
```
>>> {"test": 123} or {}
{'test': 123}
```